src/cloud-api-adaptor/pkg/podnetwork/common.go

```diff
@@ -4,42 +4,270 @@
+		// is considered the secondary interface.
+
+		if !addr.IsValid() {
+			fmt.Printf("Skipping interface %q as it has no valid IPv4 address\n", link.Name())
+			continue
+		}
+
+		if route != nil {
+			fmt.Printf("Secondary interface %q found with IPv4 address %q and route %v\n", link.Name(), addr, route)
+			return link.Name(), addr, route, nil
+		}
+
+		fmt.Printf("Route is nil, checking if %q is in the same subnet as %q\n", addr, primaryInterface)
+		priAddrCIDR, defRoute, err := getInterfaceDetails(ns, primaryInterface)
+		if err != nil {
+			return "", netip.Prefix{}, nil, err
+		}
+		if priAddrCIDR.Bits() == addr.Bits() {
+			secIface = link.Name()
+			secRoute = &netops.Route{
+				Destination: defRoute.Destination,
+				Gateway:     defRoute.Gateway,
+				Device:      secIface,
+				// Change the Route.Device to the secondary interface
+			}
+			fmt.Printf("Secondary interface %q found with IPv4 address %q and inferred route %v\n", link.Name(), addr, secRoute)
+			return link.Name(), addr, secRoute, nil
+		}
+
+	}
+
+	return "", netip.Prefix{}, nil, ErrNoSecondaryInterface
+}
+
+// isInterfaceFilteredOut filters out interface names that begin with certain prefixes.
+// This is used to ignore virtual, loopback, and other non-relevant interfaces.
+//
+// The prefixes to filter out are:
+// "veth", "lo", "docker", "podman", "br-", "cni", "tunl", "tun", "tap"
+var allowedPrefixes = []string{
+	"veth", "lo", "docker", "podman", "br-", "cni", "tunl", "tun", "tap",
+}
+
+func isInterfaceFilteredOut(ifName string) bool {
+
+	for _, prefix := range allowedPrefixes {
+		if strings.HasPrefix(ifName, prefix) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Function to move the network interface to a different network namespace
+// Also set the default route and address for the interface in the new namespace
+
+func moveInterfaceToNamespace(srcNs, dstNs netops.Namespace, iface string, addrCIDR netip.Prefix, defRoute *netops.Route) error {
+
+	// Get the network interface object
+	link, err := srcNs.LinkFind(iface)
+	if err != nil {
+		return fmt.Errorf("failed to get link %q: %w", iface, err)
+	}
+
+	// Move the network interface to the new namespace
+	err = link.SetNamespace(dstNs)
+	if err != nil {
+		return fmt.Errorf("failed to move link %q to namespace %q: %w", iface, dstNs.Path(), err)
+	}
+
+	// Set the address for the network interface in the new namespace
+	err = link.AddAddr(addrCIDR)
+	if err != nil {
+		return fmt.Errorf("failed to set address %q for link %q: %w", addrCIDR, iface, err)
+	}
+
+	// Bring up the network interface in the new namespace
+	err = link.SetUp()
+	if err != nil {
+		return fmt.Errorf("failed to bring up link %q: %w", iface, err)
+	}
+
+	// Get existing default route in the new namespace
+	defRoutes, err := dstNs.GetDefaultRoutes()
+	if err != nil {
+		return fmt.Errorf("failed to get default route in namespace %q: %w", dstNs.Path(), err)
```

Comment:
Same here? in case the loop could finish without finding the default routes? 

Some explicit checks like this?

```
if len(defRoutes) == 0 {
	return nil, fmt.Errorf("no default route found")
}
```

Reply:
defRoutes is a slice and will return empty slice if no defRoutes exist. So no need to recheck it again.